sklearn/neighbors/_base.py

```diff
@@ -749,7 +756,8 @@ def kneighbors_graph(self, X=None, n_neighbors=None,
         Parameters
         ----------
         X : array-like, shape (n_queries, n_features), \
-                or (n_queries, n_indexed) if metric == 'precomputed'
+                or (n_queries, n_indexed) if metric == 'precomputed', \
+                default=None
             The query point or points.
             If not provided, neighbors of each indexed point are returned.
             In this case, the query point is not considered its own neighbor.
```

Comment:
```suggestion
                or (n_queries, n_indexed), default=None
            The query point or points.
            If not provided, neighbors of each indexed point are returned.
            In this case, the query point is not considered its own neighbor.
            For ``metric='precomputed'`` the shape should be
            (n_queries, n_indexed). Otherwise the shape should be
            (n_queries, n_features).
```

Reply:
I don't know if it is preferred to place the description of when to use which shape in the parameter description instead of in the title. But I have often seen it like this in the documentation.